source/MaterialXCore/UnitConverter.cpp

```diff
@@ -127,15 +130,30 @@ Vector4 LengthUnitConverter::convert(Vector4 input, const string& inputUnit, con
     return (input * conversionRatio(inputUnit, outputUnit));
 }
 
-/// Given a unit name return a value that it can map to as an integer
 int LengthUnitConverter::getUnitAsInteger(const string& unitName) const
 {
-    const auto it = _unitScale.find(unitName);
```

Comment:
Added an explicit unit  enumeration as the storage order is not guaranteed to be deterministic the for a map.